src/sandbox/interpolation/scripIO.cc

```diff
@@ -0,0 +1,264 @@
+    {
+        std::vector<size_t> sorted(nnz);
+        for (size_t n = 0; n < nnz; ++n) {
+            sorted[n] = n;
+        }
+
+        std::sort(sorted.begin(), sorted.end(), [&](auto i, auto j) {
+            return ia[i] != ia[j] ? ia[i] < ia[j] : ja[i] != ja[j] ? ja[i] < ja[j] : i < j;
+        });
+
+        decltype(ia) ia_unsorted(Nr + 1, 0);
+        decltype(ja) ja_unsorted(nnz);
+        ia.swap(ia_unsorted);
+        ja.swap(ja_unsorted);
+
+        const auto base = static_cast<Index>(1);
+        for (size_t n = 0; n < nnz; ++n) {
+            auto r = ia_unsorted[sorted[n]] - base;
+            auto c = ja_unsorted[sorted[n]] - base;
+            ATLAS_ASSERT(0 <= r && r < Nr);
+            ATLAS_ASSERT(0 <= c && c < Nc);
+
+            ia[r + 1]++;
+            ja[n] = c;
+        }
+
+        ia_unsorted.clear();
+        ja_unsorted.clear();
+
+        for (size_t r = 0; r < Nr; ++r) {
+            ia[r + 1] += ia[r];
+        }
+
+        ATLAS_ASSERT(ia.front() == 0 && ia.back() == nnz);
+
+        decltype(a) a_unsorted(nnz);
+        a.swap(a_unsorted);
+
+        for (size_t n = 0; n < nnz; ++n) {
+            a[n] = a_unsorted[sorted[n]];
+            ATLAS_ASSERT(0. <= a[n] && a[n] <= 1.);
+        }
+    }
+
+    eckit::linalg::SparseMatrix eckit_matrix(new InPlaceAllocator(Nr, Nc, nnz, ia.data(), ja.data(), a.data()));
+    return atlas::linalg::make_sparse_matrix_storage(eckit_matrix);
+#endif // ATLAS_HAVE_NETCDF
+}
+
+
+void ScripIO::write(const SparseMatrixStorage& matrix, const std::string& matrix_name) {
+#if ATLAS_HAVE_NETCDF == 0
+    ATLAS_THROW_EXCEPTION("Cannot write SCRIP files: Atlas not compiled with NetCDF support");
+#else
+    using scrip_index = int;
+    using scrip_value = double;
+    using scrip_size  = size_t;
+    scrip_size Nr  = matrix.rows();
+    scrip_size Nc  = matrix.cols();
+    scrip_size nnz = matrix.nnz();
+
+    std::vector<scrip_index> ia(nnz);
+    std::vector<scrip_index> ja(nnz);
+    std::vector<scrip_value> a(nnz);
+
+    matrix_to_rows_columns_values(matrix, ia, ja, a);
+    for (size_t i = 0; i < matrix.nnz(); ++i) {
+        ++ja[i];
+        ++ia[i];
+    }
+
+    try {
+        netCDF::NcFile f(matrix_name, netCDF::NcFile::replace);
+
+        netCDF::NcDim xdim = f.addDim("dst_grid_size", Nr);
+        netCDF::NcDim ydim = f.addDim("src_grid_size", Nc);
+
+        std::vector<netCDF::NcDim> dims;
+        dims.push_back(f.addDim("num_links", nnz));
+        netCDF::NcVar nc_dstaddr = f.addVar("dst_address", netCDF::ncInt, dims);
+        netCDF::NcVar nc_srcaddr = f.addVar("src_address", netCDF::ncInt, dims);
+        dims.push_back(f.addDim("num_wgts", 1));
+        netCDF::NcVar nc_rmatrix = f.addVar("remap_matrix", netCDF::ncDouble, dims);
+
+        nc_dstaddr.putVar(ia.data());
+        nc_srcaddr.putVar(ja.data());
+        nc_rmatrix.putVar(a.data());
+        f.close();
+    }
+    catch (netCDF::exceptions::NcException& e) {
+        e.what();
```

Comment:
This catch currently means that error is printed but then ignored.
```suggestion
        ATLAS_THROW_EXCEPTION(e.what());
```

Reply:
Indeed. Thanks!